android/src/main/kotlin/com/apptentive/apptentive_flutter/ApptentiveFlutterPlugin.kt

```diff
@@ -286,74 +302,108 @@ class ApptentiveFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAware
   // Register listeners for native callbacks:
   private fun registerListeners(result: Result) {
     try {
-      Apptentive.setOnSurveyFinishedListener { channel.invokeMethod("onSurveyFinished", mapOf("completed" to it)) }
-      Apptentive.addUnreadMessagesListener { channel.invokeMethod("onUnreadMessageCountChanged", mapOf("count" to it)) }
+      registerForSurveyListener()
+      registerForUnreadMessageListener()
       result.success(true)
     } catch(e: Exception) {
       result.error(ERROR_CODE, "Failed to register Apptentive listeners.", e.toString())
     }
   }
 
+  private fun registerForSurveyListener() {
+    Apptentive.eventNotificationObservable.observe { notification ->
+      val name = notification?.name
+      val interaction = notification?.interaction
+      val vendor = notification?.vendor
+      val interactionId = notification?.interactionId
+      val notificationText = "Name: \"$name\". Vendor: \"$vendor\". " +
+              "Interaction: \"$interaction\". Interaction ID: $interactionId"
+      Log.d(LogTags.EVENT_NOTIFICATION, notificationText)
+      when {
+        interaction == "Survey" && name == "submit" ->
+          activity?.runOnUiThread {
+            channel.invokeMethod("onSurveyFinished", mapOf("completed" to true))
+          }
+
+        interaction == "Survey" && name != "submit" ->
+          activity?.runOnUiThread {
+            channel.invokeMethod("onSurveyFinished", mapOf("completed" to false))
```

Comment:
Do we have to invoke when a survey is not completed?

Reply:
Legacy plugin did that. Keeping the parity.